app/src/main/java/de/westnordost/streetcomplete/screens/about/LogsFragment.kt

```diff
@@ -0,0 +1,145 @@
+import de.westnordost.streetcomplete.data.logs.format
+import de.westnordost.streetcomplete.databinding.FragmentLogsBinding
+import de.westnordost.streetcomplete.screens.TwoPaneDetailFragment
+import de.westnordost.streetcomplete.util.ktx.now
+import de.westnordost.streetcomplete.util.ktx.toEpochMilli
+import de.westnordost.streetcomplete.util.ktx.viewLifecycleScope
+import de.westnordost.streetcomplete.util.viewBinding
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import kotlinx.datetime.LocalDateTime
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import org.koin.android.ext.android.inject
+
+/** Shows the app logs */
+class LogsFragment : TwoPaneDetailFragment(R.layout.fragment_logs) {
+
+    private val logsController: LogsController by inject()
+    private val binding by viewBinding(FragmentLogsBinding::bind)
+    private val adapter = LogsAdapter()
+
+    private var filters = LogsFilters()
+
+    private val logsControllerListener = object : LogsController.Listener {
+        override fun onAdded(message: LogMessage) { viewLifecycleScope.launch { onMessageAdded(message) } }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        createOptionsMenu(binding.toolbar.root)
+
+        binding.logsList.adapter = adapter
+        binding.logsList.itemAnimator = null // default animations are too slow when logging many messages quickly
+        binding.logsList.addItemDecoration(DividerItemDecoration(context, DividerItemDecoration.VERTICAL))
+
+        if (savedInstanceState != null) {
+            onLoadInstanceState(savedInstanceState)
+        }
+
+        showLogs()
+
+        logsController.addListener(logsControllerListener)
+    }
+
+    private fun onLoadInstanceState(savedInstanceState: Bundle) {
+        filters = Json.decodeFromString(savedInstanceState.getString(FILTERS_DATA)!!)
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        outState.putString(FILTERS_DATA, Json.encodeToString(filters))
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        logsController.removeListener(logsControllerListener)
+    }
+
+    private fun createOptionsMenu(toolbar: Toolbar) {
+        toolbar.inflateMenu(R.menu.menu_logs)
+
+        toolbar.setOnMenuItemClickListener { item ->
+            when (item.itemId) {
+                R.id.action_share -> {
+                    onClickShare()
+                    true
+                }
+                R.id.action_filter -> {
+                    onClickFilter()
+                    true
+                }
+                else -> false
+            }
+        }
+    }
+
+    private fun onClickShare() = viewLifecycleScope.launch {
+        val shareIntent = Intent(Intent.ACTION_SEND).apply {
```

Comment:
By the way, I usually avoid `.apply` and other functions whose block redefines `this` at least within classes, as one often needs the IDE's code hints to avoid the confusion what `this` refers to.

But anyway, not a hard rule and certainly not done consistently throughout the codebase, so I don't mind this here.